File: Bookworm Backend Dot net/Models/ProductType.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Bookworm.Models
{
    /// </summary>
    [Table("product_type_master")]
    public class ProductType
    {
        [Key]
        [Column("type_id")]
        public int Id { get; set; }

        [Required]
        [StringLength(255)]
        [Column("type_desc")]
        public string Description { get; set; }
    }
}

File: Bookworm Backend Spring Boot/src/main/java/com/bookworm/entities/ProductType.java
package com.bookworm.entities;

import java.math.BigDecimal;
import java.util.Set;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

/**
 * Represents the type of a product (e.g., E-Book, Audiobook).
 * Maps to the 'product_types' table.
 */
@Entity
@Table(name = "product_type_master")
@Getter
@Setter
public class ProductType {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "type_id")
    private Integer id;

    @Column(name = "type_desc", length = 255, nullable = false)
    private String description;

   
}
